test/battle/ability/clear_body.c

```diff
@@ -26,13 +26,211 @@ SINGLE_BATTLE_TEST("Clear Body prevents intimidate")
+            }
+    }
+}
+
+SINGLE_BATTLE_TEST("Clear Body doesn't prevent Attack reduction from burn", s16 damage)
+{
+    bool32 burned;
+    PARAMETRIZE{ burned = FALSE; }
+    PARAMETRIZE{ burned = TRUE; }
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_TACKLE].category == DAMAGE_CATEGORY_PHYSICAL);
+        PLAYER(SPECIES_WOBBUFFET)
+        OPPONENT(SPECIES_BELDUM) { Ability(ABILITY_CLEAR_BODY); if (burned) Status1(STATUS1_BURN); }
+    } WHEN {
+        TURN{ MOVE(opponent, MOVE_TACKLE); }
+    } SCENE {
+        HP_BAR(player, captureDamage: &results[i].damage);
+    } FINALLY {
+        EXPECT_MUL_EQ(results[0].damage, Q_4_12(0.5), results[1].damage);
+    }
+}
+
+SINGLE_BATTLE_TEST("Clear Body doesn't prevent receiving negative stat changes from Baton Pass")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_SCARY_FACE].effect == EFFECT_SPEED_DOWN_2);
+        ASSUME(gMovesInfo[MOVE_BATON_PASS].effect == EFFECT_BATON_PASS);
+        PLAYER(SPECIES_WOBBUFFET) { Speed(4); Moves(MOVE_SCARY_FACE); }
+        OPPONENT(SPECIES_WOBBUFFET) { Speed(3); Moves(MOVE_BATON_PASS); }
+        OPPONENT(SPECIES_BELDUM) { Speed(6); Ability(ABILITY_CLEAR_BODY); }
+    } WHEN {
+        TURN{ MOVE(player, MOVE_SCARY_FACE); MOVE(opponent, MOVE_BATON_PASS); SEND_OUT(opponent, 1); }
+        TURN{ MOVE(player, MOVE_SCARY_FACE); MOVE(opponent, MOVE_CELEBRATE); }
+    } SCENE {
+            MESSAGE("Wobbuffet used Scary Face!");
+            MESSAGE("Foe Beldum used Celebrate!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Clear Body doesn't prevent Topsy-Turvy")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_TOPSY_TURVY].effect == EFFECT_TOPSY_TURVY);
+        ASSUME(gMovesInfo[MOVE_SCARY_FACE].effect == EFFECT_SPEED_DOWN_2);
+        ASSUME(gMovesInfo[MOVE_BATON_PASS].effect == EFFECT_BATON_PASS);
+        PLAYER(SPECIES_WOBBUFFET) { Speed(4); Moves(MOVE_SCARY_FACE, MOVE_TOPSY_TURVY); }
+        OPPONENT(SPECIES_WOBBUFFET) { Speed(3); Moves(MOVE_BATON_PASS); }
+        OPPONENT(SPECIES_BELDUM) { Speed(6); Ability(ABILITY_CLEAR_BODY); }
+    } WHEN {
+        TURN{ MOVE(player, MOVE_SCARY_FACE); MOVE(opponent, MOVE_BATON_PASS); SEND_OUT(opponent, 1); }
+        TURN{ MOVE(player, MOVE_TOPSY_TURVY); MOVE(opponent, MOVE_CELEBRATE); }
+        TURN{ MOVE(player, MOVE_SCARY_FACE); MOVE(opponent, MOVE_CELEBRATE); }
+    } SCENE {
+            MESSAGE("Wobbuffet used Topsy-Turvy!");
+            MESSAGE("Foe Beldum used Celebrate!");
+            MESSAGE("Foe Beldum used Celebrate!");
+            MESSAGE("Wobbuffet used Scary Face!");
+    }
+}
+
+SINGLE_BATTLE_TEST("Clear Body doesn't prevent Spectral Thief from resetting positive stat changes")
+{
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_SPECTRAL_THIEF].additionalEffects->moveEffect == MOVE_EFFECT_SPECTRAL_THIEF);
+        ASSUME(gMovesInfo[MOVE_AGILITY].effect == EFFECT_SPEED_UP_2);
+        PLAYER(SPECIES_WOBBUFFET) { Speed(4); Moves(MOVE_SPECTRAL_THIEF, MOVE_CELEBRATE); }
```

Comment:
Implicit moves

Reply:
This one specifically, I can't remove the Moves(); sections without the compiler telling me the moves are invalid, in spite of the rest of the cleanup. Not sure why, let me know if I'm missing something obvious when you get to re-review :)

Reply:
@mrgriffin do you know why this might be?

Reply:
You can't remove them from both mons or does it only give an error you remove it from one mon?

Reply:
Removing from both mons or an individual mon all caused errors

Reply:
weird